ConcatenateMatrices.java

```diff
@@ -3,33 +3,69 @@
 
 public class ConcatenateMatrices {
 
-    public static void main(String[] args) throws IOException {
-        Scanner scanner = new Scanner(new File("vstup.txt"));
-        PrintStream output = new PrintStream("vystup.txt");
+    public static void main(String[] args){
 
-        int m = scanner.nextInt();
-        int n = scanner.nextInt();
+        Scanner scanner = null;
+        PrintStream output = null;
+        try {
+            scanner = new Scanner(new File("vstup.txt"));
+            output = new PrintStream("vystup.txt");
+        }
+        catch (Exception e){
+            System.out.println("Problem s nacitanim/vytvorenim vstupneho/vystupneho suboru.");
+            if(scanner != null) scanner.close();
+            if(output != null) output.close();
+            return;
+        }
+
+        int m, n;
+        try {
+            m = scanner.nextInt();
+            n = scanner.nextInt();
+            if(m < 1 || n < 1) throw new IllegalArgumentException();
+        }
+        catch (Exception e){
```

Comment:
Catching everything is generally considered a bad practice. 
https://softwareengineering.stackexchange.com/questions/164256/is-catching-general-exceptions-really-a-bad-thing